### ISSUE TITLE
Support for lifetimes

### ISSUE THREAD

**Issue by magnusart** [lffg/bty#1]
When I try to add a lifetime  generic I get a compiler error:

```rust
bty::brand!(
    pub type UserName = &'static str; // This works!
    pub type UserEmail<'a> = &'a str; // Error: no rules expected the token `<`
    // no rules expected this token in macro call
);
```

Suggestion: provide support for lifetime parameters.

Note! I'm new to rust so forgive me if this is a newbie mistake regarding the lifetime.

**Comment by lffg** [comment]
Yeah, that's indeed not currently supported, as the macro definition only accepts an identifier token, see:
https://github.com/lffg/bty/blob/48f7786329eae3cacec9cb90408b473ff63c31ab/src/lib.rs#L42

I shall commit a patch that fixes the syntax to allow lifetimes, generic parameter and trait bounds as soon as possible. Thanks for the report. :)
